src/backend/catalog/gp_matview_aux.c

```diff
@@ -462,3 +462,16 @@ MatviewIsGeneralyUpToDate(Oid mvoid)
 	return ((auxform->datastatus == MV_DATA_STATUS_UP_TO_DATE) || 
 			(auxform->datastatus == MV_DATA_STATUS_UP_REORGANIZED));
 }
+
+bool
+MatviewIsUpToDate(Oid mvoid)
+{
+	HeapTuple mvauxtup = SearchSysCacheCopy1(MVAUXOID, ObjectIdGetDatum(mvoid));
+
+	/* Not a candidate we recorded. */
+	if (!HeapTupleIsValid(mvauxtup))
+		return false;
+
+	Form_gp_matview_aux auxform = (Form_gp_matview_aux) GETSTRUCT(mvauxtup);
+	return (auxform->datastatus == MV_DATA_STATUS_UP_TO_DATE);
+}
```

Comment:
That function is differ from MatviewIsGeneralyUpToDate only in predicate  ``|| auxform->datastatus == MV_DATA_STATUS_UP_REORGANIZED``

Maybe it would be better to create function like matviewCheckForDataStatuses(Oid mvoid, char[] statuses) and not to duplicate steps to get Form_gp_matview_aux structure

Reply:
HI, @leborchuk thanks for review.

> Maybe it would be better to create function like matviewCheckForDataStatuses(Oid mvoid, char[] statuses)

I personally can't agree that,  it will make much more complex for function callers to construct a char[], then the couple of lines you saved will be back.
And, a lot of codes need to adjust with new function.